hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java

```diff
@@ -604,7 +612,9 @@ public void removeEdge(HugeEdge edge) {
     public Iterator<Edge> queryEdges(Query query) {
         Iterator<HugeEdge> results = this.queryEdgesFromBackend(query);
 
-        Set<Id> returnedEdges = new HashSet<>();
+        // TODO: any unconsidered case?
+        boolean withDuplicatedEdge = false;
```

Comment:
When will it be true?

Reply:
maybe the query with OR condition, like query {OUT edges of vertex1} OR {IN edges of vertex2}